client.cpp

```diff
@@ -332,20 +350,37 @@ PHP_METHOD(VoltClient, connect)
     }
 
     if (argc == 4) {
-        obj->client = new voltdb::Client(voltdb::ConnectionPool::pool()->acquireClient(
-                                             err, hostname,
-                                             username, password,
-                                             &status_listener, port));
+        try {
+            obj->client = new voltdb::Client(voltdb::ConnectionPool::pool()->acquireClient(
+                                                 hostname, username,
+                                                 password, &status_listener,
+                                                 port, voltdb::HASH_SHA256));
+        } catch (voltdb::ConnectException) {
+            zend_throw_exception(zend_exception_get_default(TSRMLS_C), NULL, errConnectException TSRMLS_CC);		
+            RETURN_FALSE;
+        } catch (voltdb::LibEventException) {
+            zend_throw_exception(zend_exception_get_default(TSRMLS_C), NULL, errLibEventException TSRMLS_CC);		
+            RETURN_FALSE;
+        } catch (voltdb::Exception) {
+            zend_throw_exception(zend_exception_get_default(TSRMLS_C), NULL, errException TSRMLS_CC);		
+            RETURN_FALSE;
+        }
     } else {
-        obj->client = new voltdb::Client(voltdb::ConnectionPool::pool()->acquireClient(
-                                             err, hostname,
-                                             username, password,
-                                             &status_listener));
-    }
-
-    if (!voltdb::isOk(err)) {
-        zend_throw_exception(zend_exception_get_default(TSRMLS_C), NULL, err TSRMLS_CC);
-        RETURN_FALSE;
+        try {
+            obj->client = new voltdb::Client(voltdb::ConnectionPool::pool()->acquireClient(
+                                                 hostname, username,
+                                                 password, &status_listener,
+                                                 21212, voltdb::HASH_SHA256));
```

Comment:
The original code was separated into two branches because they were calling two different flavors of the `acquireClient()` method. It seems that you are calling the same method now. You can probably consolidate the branches.